src/meson.build

```diff
@@ -0,0 +1,67 @@
+incdir = include_directories('../include/')
+
+install_headers(
+    '../include/lsst/sphgeom/Angle.h',
+    '../include/lsst/sphgeom/AngleInterval.h',
+    '../include/lsst/sphgeom/BigInteger.h',
+    '../include/lsst/sphgeom/Box3d.h',
+    '../include/lsst/sphgeom/Box.h',
+    '../include/lsst/sphgeom/Chunker.h',
+    '../include/lsst/sphgeom/Circle.h',
+    '../include/lsst/sphgeom/codec.h',
+    '../include/lsst/sphgeom/CompoundRegion.h',
+    '../include/lsst/sphgeom/constants.h',
+    '../include/lsst/sphgeom/ConvexPolygon.h',
+    '../include/lsst/sphgeom/curve.h',
+    '../include/lsst/sphgeom/Ellipse.h',
+    '../include/lsst/sphgeom/HtmPixelization.h',
+    '../include/lsst/sphgeom/Interval1d.h',
+    '../include/lsst/sphgeom/Interval.h',
+    '../include/lsst/sphgeom/LonLat.h',
+    '../include/lsst/sphgeom/Matrix3d.h',
+    '../include/lsst/sphgeom/Mq3cPixelization.h',
+    '../include/lsst/sphgeom/NormalizedAngle.h',
+    '../include/lsst/sphgeom/NormalizedAngleInterval.h',
+    '../include/lsst/sphgeom/orientation.h',
+    '../include/lsst/sphgeom/Pixelization.h',
+    '../include/lsst/sphgeom/python.h',
+    '../include/lsst/sphgeom/Q3cPixelization.h',
+    '../include/lsst/sphgeom/RangeSet.h',
+    '../include/lsst/sphgeom/Region.h',
+    '../include/lsst/sphgeom/Relationship.h',
+    '../include/lsst/sphgeom/UnitVector3d.h',
+    '../include/lsst/sphgeom/utils.h',
+    '../include/lsst/sphgeom/Vector3d.h',
+    subdir : 'lsst/sphgeom'
+)
+sphgeom_sources = [
+    'Angle.cc',
+    'AngleInterval.cc',
+    'BigInteger.cc',
+    'Box3d.cc',
+    'Box.cc',
+    'Chunker.cc',
+    'Circle.cc',
+    'CompoundRegion.cc',
+    'ConvexPolygon.cc',
+    'Ellipse.cc',
+    'HtmPixelization.cc',
+    'Interval1d.cc',
+    'LonLat.cc',
+    'Matrix3d.cc',
+    'Mq3cPixelization.cc',
+    'NormalizedAngle.cc',
+    'NormalizedAngleInterval.cc',
+    'orientation.cc',
+    'Q3cPixelization.cc',
+    'RangeSet.cc',
+    'Region.cc',
+    'UnitVector3d.cc',
+    'utils.cc',
+    'Vector3d.cc']
+sphgeom = static_library(
```

Comment:
Why a static library, particularly? If you just use `library()` it becomes the choice of the builder (via `-Ddefault_library=static`, but it also accepts `=both` to build both at the same time).

The cmake build uses shared, not static, though cmake doesn't really have a good way to choose between the two.